metpy/cbook.py

```diff
@@ -39,4 +37,103 @@ def get_test_data(fname, as_file_obj=True):
     return path
 
 
-__all__ = ('get_test_data', 'is_string_like', 'iterable')
+class CacheData(object):
+    def __init__(self):
+        self.cache_home = self.get_cache_location()
+        self.cache_version = 1
+
+        # Check if there is a cache version file. If not, this is a new cache, so create it.
+        cache_version_file_path = os.path.join(self.cache_home, 'cache_version.txt')
+        if not os.path.exists(cache_version_file_path ):
+            with open(cache_version_file_path , 'w') as f:
+                f.write('{}\n'.format(self.cache_version))
+
+        # Check if this is an old cache version. If it is, remove the cache so we are sure
+        # to be using current files
+        with open(cache_version_file_path, 'r') as f:
+            current_cache_version = int(f.readline())
+        if current_cache_version != self.cache_version:
+            self.clear_cache()
+            # Now that we've blown the old cache away, create a new one and
+            # a version file with the current cache version
+            self.cache_home = self.get_cache_location()
+            with open(cache_version_file_path, 'w') as f:
+                f.write('{}\n'.format(self.cache_version))
+
+    def get_cache_location(self):
```

Comment:
I'm thinking making this a property (as shown above) makes sense. Also, let's pull out the `makedirs` here so that it just does the *one* thing.